libbeat/publisher/pipeline/pipeline.go

```diff
@@ -349,23 +369,96 @@ func (p *Pipeline) ConnectWith(cfg beat.ClientConfig) (beat.Client, error) {
 		acker = nilACKer
 	}
 
-	producer := p.queue.Producer(producerCfg)
-	client := &client{
-		pipeline:     p,
-		isOpen:       atomic.MakeBool(true),
-		eventer:      cfg.Events,
-		processors:   processors,
-		producer:     producer,
-		acker:        acker,
-		eventFlags:   eventFlags,
-		canDrop:      canDrop,
-		reportEvents: reportEvents,
-	}
+	client.acker = acker
+	client.producer = p.queue.Producer(producerCfg)
 
 	p.observer.clientConnected()
+
+	if client.closeRef != nil {
+		p.registerSignalPropagation(client)
+	}
+
 	return client, nil
 }
 
+func (p *Pipeline) registerSignalPropagation(c *client) {
+	p.guardStartSigPropagation.Do(func() {
+		p.sigNewClient = make(chan *client, 1)
+		go p.runSignalPropagation()
+	})
+	p.sigNewClient <- c
+}
+
+func (p *Pipeline) runSignalPropagation() {
+	var channels []reflect.SelectCase
```

Comment:
TIL, never use that before.

Reply:
I need to become more experienced with the reflect package, I don't use it that much.